d2bs/kolbot/libs/modules/Town.js

```diff
@@ -352,6 +352,31 @@
 		return true;
 	};
 
+	// Purchases and drinks potions in town.
+	Town.stackPotions = function(type, amount) {
+		if (!amount) amount = 4; // TODO: Calculate monster effort and correlate it to the number of potions to stack.
```

Comment:
TODO: _I think that developing these functions is outside the scope of the ticket, but eventually we'll want to determine the amount of potions dynamically._

Boss/Area Effort **BE** can be broken down into the "average spell casts" described in `AutoMagicFind.js` -- but we'll probably want to break that out into it's own function and maybe cache/memoize the results.
Spell Average Cast Time **SACT** is the average time it takes a spell to cast. This can be estimated by determining FCR and spell timed/untimed events. Or just assigned an appropriate arbitrary number (like `0.3`).
Potion Length **PL** is always 30 seconds (per potion).

`amount = ceil((BE * SACT) / PL)`

`amount` should then be the average number of potions needed to clear the area with 100% potion uptime.